spec/Component.coffee

```diff
@@ -1636,6 +1636,103 @@ describe 'Component', ->
         sin1.post new noflo.IP 'data', 'moose'
         sin1.post new noflo.IP 'closeBracket'
 
+    describe 'with a simple ordered stream', ->
+      it 'should send packets with brackets in expected order when synchronous', (done) ->
+        received = []
+        c = new noflo.Component
+          inPorts:
+            in:
+              datatype: 'string'
+          outPorts:
+            out:
+              datatype: 'string'
+          process: (input, output) ->
+            return unless input.has 'in'
```

Comment:
So checking that port had input data is now mandatory, even for the simplest cases?